src/Adapter/Grid/Search/Factory/SearchCriteriaWithCategoryParentIdFilterFactory.php

```diff
@@ -68,25 +69,33 @@ final class SearchCriteriaWithCategoryParentIdFilterFactory implements Decorated
      */
     private $contextShopCategoryId;
 
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
     /**
      * @param Configuration $configuration
      * @param Context $shopContext
      * @param FeatureInterface $multistoreFeature
      * @param MultistoreContextCheckerInterface $multistoreContextChecker
      * @param int $contextShopCategoryId
+     * @param RequestStack $requestStack
      */
     public function __construct(
         Configuration $configuration,
         Context $shopContext,
         FeatureInterface $multistoreFeature,
         MultistoreContextCheckerInterface $multistoreContextChecker,
-        $contextShopCategoryId
+        $contextShopCategoryId,
+        RequestStack $requestStack
```

Comment:
I was not allowed to do that. :expressionless:

Reply:
Because it's a bad thing to couple a service with the RequestStack 😄and indeed if we can avoid it, I'd feel better

It looks like the recursive nature of Categories is polluting our code 😢

Reply:
I believe I remember we told you not to use it, but in the en using Tools instead of the request is kind of the same
I actually tried to modify the least possible things in your code, but I found it quite complicated..
There's a listener class which then calls for a factory decorator.. this seems a bit overkill when the listener could have done the job by its own

Reply:
@sarjon we discussed about this with @matks 
Neither of us satisfied with the use of request stack in this service.. we would like to refacto this behaviour for specific Filters
However considering the left time before release it seems dangerous to perform a big change, so sadly we are considering merging this fix for 176 (as long as it works correctly and is validated by QA) but we plan on a refacto in 177
I created an issue about this subject explaining an idea I had based on the `FiltersBuilder` https://github.com/PrestaShop/PrestaShop/issues/14113

Reply:
@jolelievre thats totally fine, it's not your problem, but global issue in PS that I think that we all need to improve. :slightly_smiling_face:

Reply:
Yep I agree with you, so sad I couldn't get it right for 176 😭
Feel free to comment the issue I created if you have feedbacks